denizen_scripts/survival/repo-link/items/system/soul_item_system.dsc

```diff
@@ -247,58 +337,78 @@ item_with_soul:
-    - if <[buffs]||null> != null:
-      - define buffs_and_debuffs:|:buffs/<[buffs].escaped>
-    - if <[debuffs]||null> != null:
-      - define buffs_and_debuffs:|:debuffs/<[debuffs].escaped>
-    - define flavor:<element[<&d>Soul&spItem<&nl><&e>Combine&spwith&spArmor&spor&spWeapons].escaped>
-    - determine <proc[item_system_build_item].context[<item[soul].with[nbt=soul/<[soul_type]>,<[level]>|rarity/<[rarity]>|soul_level/<[level]>|active_soul/<[soul_type]>|flavor/<[flavor]>|<[buffs_and_debuffs].separated_by[|]>]>]>
+        - define debuffs:->:<map.with[<[stat]>].as[<script[item_system_global_data].parsed_key[calculations.<[stat]>]>]>
+    - define buffs_and_debuffs <map>
+    - if <[buffs].exists>:
+      - define buffs_and_debuffs <[buffs_and_debuffs].with[buffs].as[<list_single[<[buffs]>]>]>
+    - if <[debuffs].exists>:
+      - define buffs_and_debuffs <[buffs_and_debuffs].with[debuffs].as[<list_single[<[debuffs]>]>]>
+    - define flavor "<&d>Soul Item<&nl><&e>Combine with Armor or Weapons"
+    - define NBT <[NBT].with[flavor].as[<[flavor]>].include[<[buffs_and_debuffs]>]>
+    - determine <proc[item_system_build_item].context[<item[soul].with[nbt=<[NBT]>]>]>
 
 item_with_soul_create:
   type: procedure
   definitions: item1|item2
   script:
     - if <[item1].material.name> == air || <[item2].material.name> == air:
       - determine gray_stained_glass_pane
-    - if !<list[<[item1]>|<[item2]>].filter[material.name.is[==].to[<script[soul].yaml_key[material]>]].get[1].has_nbt[soul]>:
+    - if !<list[<[item1]>].include[<[item2]>].filter[material.name.is[==].to[<script[soul].data_key[material]>]].first.has_nbt[soul]>:
       - determine gray_stained_glass_pane
-    - define soul_item <list[<[item1]>|<[item2]>].filter[has_nbt[soul]].get[1]>
-    - define soul_type <[soul_item].nbt[soul].before[,]>
-    - define soul_level <[soul_item].nbt[soul].after[,]>
-    - define rarity <[soul_item].nbt[rarity]>
-    - define item <list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].get[1].material.name>
-    - define enchants <list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].get[1].enchantments.with_levels.escaped||<list[]>>
-    - determine <proc[item_create_soul_item].context[<[item]>|<[soul_type]>|<[soul_level]>|<[rarity]>|<[enchants]>]>
+
+    - define soul_item <list[<[item1]>].include[<[item2]>].filter[has_nbt[soul]].first>
+    - define contexts <list.include[<list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].first.material.name>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[soul].get[type]>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[soul].get[level]>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[rarity]>]>
+    - define contexts <[context].include[<list_single[<list[<[item1]>|<[item2]>].parse_tag[<[parse_value].enchantments.with_levels>].combined.deduplicate>]>]>
+    - determine <proc[item_create_soul_item].context[<[Contexts]>]>
```

Comment:
removed `.escaped`/`.unescaped` as all contents are properly preserved now, used a proper list constructor as well for passing the entire list properly through as context for the proc